clients/src/main/java/org/apache/kafka/common/record/Record.java

```diff
@@ -100,149 +103,17 @@ public Record(ByteBuffer buffer) {
     }
 
     // Package private constructor for inner iteration.
```

Comment:
The wrapper values can be null if either the magic is 0 or if the record is uncompressed. I'll add an assertion for this, but I'd prefer to keep the current names since it's otherwise harder to explain.